lib/kiev/shoryuken/context_reader.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+require "kiev/context_reader"
+
+module Kiev
+  module Shoryuken
+    class ContextReader < Kiev::ContextReader
+      def [](key)
+        attributes = subject.message_attributes
```

Comment:
I am sure you have a wrong subject here. The subject should be `message_attributes`.
`context_reader = Kiev::Shoryuken::ContextReader.new(message.message_attributes)`

Reply:
No, it was a conscious choice.
This family of classes by design accepts an object representing *the whole job* with the intent that non-message-attributes-based implementation can be provided anc configured for use in the future.

Reply:
The fact that it calls `message.message_attributes` every time is a problem however. Fixed it.